auth.go

```diff
@@ -34,7 +34,7 @@ type device struct {
 type auth struct {
 	Client       *http.Client
 	AuthURL      string
-	SelectDevice func(ctx context.Context, devices []device) (d device, passcode string, err error)
+	SelectDevice func(ctx context.Context, devices []Device) (d Device, passcode string, err error)
 	userAgent    string
 }
 
```

Comment:
maybe not for this PR, but we should probably allow the caller to set a HTTP client for this